krm-functions/nfdeploy-fn/common/mutator.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2023 The Nephio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"reflect"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	infrav1alpha1 "github.com/nephio-project/api/infra/v1alpha1"
+	nephiodeployv1alpha1 "github.com/nephio-project/api/nf_deployments/v1alpha1"
+	nephioreqv1alpha1 "github.com/nephio-project/api/nf_requirements/v1alpha1"
+	kptcondsdk "github.com/nephio-project/nephio/krm-functions/lib/condkptsdk"
+	"github.com/nephio-project/nephio/krm-functions/lib/kubeobject"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func Run[T any, PT PtrIsNFDeployemnt[T]](rl *fn.ResourceList, gvk schema.GroupVersionKind) (bool, error) {
+	nfDeployFn := NewMutator[T, PT](gvk)
+
+	var err error
+
+	kptfile := rl.Items.GetRootKptfile()
```

Comment:
Should we check for nil value for kptFile? It's unlikely though it's come this far.

Reply:
Thanks for pointing out. Fixed it.